src/wayland/base/wayland_base.h

```diff
@@ -28,6 +28,11 @@ namespace mir
 namespace wayland
 {
 
+template<int V>
+struct Version
+{
+};
+
```

Comment:
I'd make this template a member of `Resource` (and `Global`) - it doesn't really have a use in its own right, and it simplifies the user code: `s/mw::Version/Version`.